File: base_module/src/main/java/kt/module/base_module/data/RvData.java
package kt.module.base_module.data;

import android.os.Parcel;
import android.os.Parcelable;

public class RvData implements Parcelable {
    private String name;
    private int age;

    public RvData() {
    }

    public RvData(String name, int age) {
        this.name = name;
        this.age = age;
    }

    @Override
    public String toString() {
        return "RvData{" +
                "name='" + name + '\'' +
                ", age=" + age +
                '}';
    }

    protected RvData(Parcel in) {
        name = in.readString();
        age = in.readInt();
    }

    public static final Creator<RvData> CREATOR = new Creator<RvData>() {
        @Override
        public RvData createFromParcel(Parcel in) {
            return new RvData(in);
        }

        @Override
        public RvData[] newArray(int size) {
            return new RvData[size];
        }
    };

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeInt(age);
    }
}

File: base_module/src/main/java/kt/module/base_module/data/RvData.kt
package kt.module.base_module.data

import android.os.Parcel
import android.os.Parcelable

class RvData : Parcelable {
    var name: String? = null
    var age: Int = 0

    constructor() {}

    constructor(name: String, age: Int) {
        this.name = name
        this.age = age
    }

    override fun toString(): String {
        return "RvData{" +
                "name='" + name + '\''.toString() +
                ", age=" + age +
                '}'.toString()
    }

    protected constructor(`in`: Parcel) {
        name = `in`.readString()
        age = `in`.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeInt(age)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<RvData> = object : Parcelable.Creator<RvData> {
            override fun createFromParcel(`in`: Parcel): RvData {
                return RvData(`in`)
            }

            override fun newArray(size: Int): Array<RvData?> {
                return arrayOfNulls(size)
            }
        }
    }
}